ons-spark/spark-functions/logistic-regression.md

```diff
@@ -0,0 +1,1660 @@
+                                      originofcall )) %>%
+  sparklyr::ft_string_indexer(input_col = "originofcall", 
+                              output_col = "originofcall_idx",
+                              string_order_type = "alphabetDesc") %>%
+  sparklyr::ft_one_hot_encoder(input_cols = c("originofcall_idx"), 
+                               output_cols = c("originofcall_ohe"), 
+                               drop_last = TRUE) %>%
+  sparklyr::sdf_separate_column(column = "originofcall_ohe",
+                                into = c("originofcall_Ambulance", 
+                                         "originofcall_Police", 
+                                         "originofcall_Coastguard",
+                                         "originofcall_Person (land line)",
+                                         "originofcall_Not Known",
+                                         "originofcall_Person (running Call)",
+                                         "originofcall_Other Frs")) 
+# propertycategory
+rescue_cat_ohe <- rescue_cat_ohe %>%
+  dplyr::mutate(propertycategory = ifelse(propertycategory == "Dwelling",
+                                          "000_Dwelling",
+                                          propertycategory)) %>%
+  sparklyr::ft_string_indexer(input_col = "propertycategory", 
+                              output_col = "propertycategory_idx",
+                              string_order_type = "alphabetDesc") %>%
+  sparklyr::ft_one_hot_encoder(input_cols = c("propertycategory_idx"), 
+                               output_cols = c("propertycategory_ohe"), 
+                               drop_last = TRUE) %>%
+  sparklyr::sdf_separate_column(column = "propertycategory_ohe",
+                                into = c("propertycategory_Outdoor", 
+                                         "propertycategory_Road Vehicle", 
+                                         "propertycategory_Non Residential",
+                                         "propertycategory_Boat",
+                                         "propertycategory_Outdoor Structure",
+                                         "propertycategory_Other Residential"))
+
+
+# remove _idx and _ohe intermediate columns and original data columns
+rescue_cat_ohe <- rescue_cat_ohe %>% 
+  sparklyr::select(-ends_with(c("_ohe", "_idx")),
+                   -originofcall,
+                   -propertycategory)
+```
+````
+
+Once this is done we can run the regression again and get the
+coefficients relative to the chosen reference categories:
+
+````{tabs}
+```{code-tab} r R 
+# Run regression with one-hot encoded variables with chosen reference categories
+glm_ohe <- dparklyr::ml_generalized_linear_regression(rescue_cat_ohe, 
+                                                      formula = "is_cat ~ .", 
+                                                      family = "binomial", 
+                                                      link = "logit")
+
```

Comment:
Assuming this is a typo and it should be sparklyr
This line gives a runtime error: java.lang.IllegalArgumentException: requirement failed: The input
  column stridx_2b446fd9d09f should have at least two distinct values.

Reply:
Input characters are of the correct type so something is wrong with the is_cat column in the input data rescue_cat_ohe:
rescue_cat_ohe$is_cat returns NULL